build/build-cdt-strings.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * @license Copyright 2022 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import fs from 'fs';
+
+import {LH_ROOT} from '../root.js';
+
+// eslint-disable-next-line max-len
+const inFile = `${LH_ROOT}/node_modules/chrome-devtools-frontend/front_end/models/issues_manager/DeprecationIssue.ts`;
+const outFile = `${LH_ROOT}/lighthouse-core/lib/deprecations-strings.js`;
+
+const code = fs.readFileSync(inFile, 'utf-8');
+
+/**
+ * @param {string} text
+ * @param {string|RegExp} searchValue
+ * @param {string} replaceValue
+ */
+function doReplacement(text, searchValue, replaceValue) {
+  const newValue = text.replace(searchValue, replaceValue);
+  if (newValue === text) throw new Error(`could not find: ${searchValue}`);
+  return newValue;
+}
+
+/**
+ * @param {string} text
+ * @param {string} needle
```

Comment:
Is `needle` common nomenclature for this? I would prefer `startPattern` and `endPattern`